test/apps/sched/sched_attr_idle.c

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#define _GNU_SOURCE
+#include <pthread.h>
+#include <sched.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+
+#define SCHED_IDLE 5
+
+pthread_barrier_t barrier;
+
+void *thread_func(void *arg)
+{
+	int thread_num = *((int *)arg);
+	struct sched_param param = { .sched_priority = 0 };
+
+	printf("Thread %d waiting at barrier\n", thread_num);
+	pthread_barrier_wait(&barrier);
+
+	int ret = sched_setscheduler(0, SCHED_IDLE, &param);
+
+	if (ret != 0) {
+		printf("Thread %d: sched_setscheduler failed: %s\n", thread_num,
+		       strerror(errno));
+	} else {
+		printf("Thread %d: successfully set to IDLE policy\n",
+		       thread_num);
+	}
+
+	sleep(1);
+	return NULL;
+}
+
+int main()
+{
+	int num_threads = 4;
+
+	if (pthread_barrier_init(&barrier, NULL, num_threads) != 0) {
+		perror("pthread_barrier_init failed");
+		exit(1);
+	}
+
+	pthread_t *threads = malloc(num_threads * sizeof(pthread_t));
+	int *thread_nums = malloc(num_threads * sizeof(int));
+
+	if (!threads || !thread_nums) {
+		perror("Memory allocation failed");
+		exit(1);
+	}
+
+	for (int i = 0; i < num_threads; i++) {
+		thread_nums[i] = i;
+		if (pthread_create(&threads[i], NULL, thread_func,
+				   &thread_nums[i]) != 0) {
+			perror("pthread_create failed");
+			exit(1);
+		}
+	}
+
+	for (int i = 0; i < num_threads; i++) {
+		pthread_join(threads[i], NULL);
+	}
+
+	printf("All threads completed\n");
+
+	pthread_barrier_destroy(&barrier);
```

Comment:
This regression test should be greatly simplified. The bug reproducing code given by #2153 is generated automatically by a fuzzer. Most of the code in this file is not essential to reproduce the bug.